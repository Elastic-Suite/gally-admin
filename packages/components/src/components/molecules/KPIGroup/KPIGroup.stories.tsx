import React from 'react'
import { ComponentMeta, ComponentStory } from '@storybook/react'

import KPIGroup from './KPIGroup'

export default {
  title: 'Molecules/KPIGroup',
  component: KPIGroup,
} as ComponentMeta<typeof KPIGroup>

const Template: ComponentStory<typeof KPIGroup> = (args) => (
  <KPIGroup {...args} />
)

export const Default = Template.bind({})

Default.args = {
  kpis: [
    {
      id: 1,
      label: 'searches',
      value: 68116,
    },
    {
      id: 2,
      label: 'sessions',
      value: 139271,
    },
    {
      id: 3,
      label: 'sessions with search',
      value: 28.6,
      isPercentage: true,
    },
    {
      id: 4,
      label: 'search per session',
      value: 1.7,
    },
    {
      id: 5,
      label: 'session with search',
      value: 28.6,
      isPercentage: true,
    },
    {
      id: 6,
      label: 'searches',
      value: 68116,
    },
  ],
}
