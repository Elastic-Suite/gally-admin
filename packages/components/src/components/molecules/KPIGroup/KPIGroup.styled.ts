import { styled } from '@mui/system'

export const Container = styled('div')(({ theme }) => ({
  display: 'flex',
  alignItems: 'center',
  flexWrap: 'wrap',
  columnGap: theme.spacing(8),
  rowGap: theme.spacing(14),
  width: '100%',
}))
