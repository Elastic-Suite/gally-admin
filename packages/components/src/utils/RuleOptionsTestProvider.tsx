import React, { ReactNode } from 'react'
import {
  IRuleEngineOperators,
  RuleAttributeType,
  getIri,
} from '@elastic-suite/gally-admin-shared'

import ruleEngineOperator from '../../public/mocks/rule_engine_operators.json'
import sourceFields from '../../public/mocks/source_fields.json'
import sourceFieldLabels from '../../public/mocks/source_field_labels.json'

import RuleOptionsProvider from '../components/stateful-providers/RuleOptionsProvider/RuleOptionsProvider'

interface IProps {
  children: ReactNode
}

const sourceFieldLabelsMap = new Map(
  sourceFieldLabels['hydra:member'].map((label) => [
    label.sourceField,
    label.label,
  ])
)

const fields = sourceFields['hydra:member'].map((field) => {
  const label = sourceFieldLabelsMap.get(getIri('source_fields', field.id))
  return {
    id: field.id,
    code: field.code,
    label: (label || field.defaultLabel) ?? '',
    type: field.type as RuleAttributeType,
  }
})

function RuleOptionsTestProvider(props: IProps): JSX.Element {
  const { children } = props
  return (
    <RuleOptionsProvider
      catalogId={-1}
      localizedCatalogId={-1}
      fields={fields}
      ruleOperators={ruleEngineOperator as IRuleEngineOperators}
    >
      {children}
    </RuleOptionsProvider>
  )
}

export default RuleOptionsTestProvider
