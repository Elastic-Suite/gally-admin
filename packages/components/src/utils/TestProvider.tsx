import React, { ReactNode } from 'react'

import catalogs from '../../public/mocks/catalog.json'
import metadata from '../../public/mocks/metadata.json'

import { catalogContext, metadataContext } from '../contexts'
import { useCatalogs } from '../hooks'

import OptionsProvider from '../components/stateful-providers/OptionsProvider/OptionsProvider'

interface IProps {
  children: ReactNode
}

function TestProvider(props: IProps): JSX.Element {
  const { children } = props
  const contextValueCatalog = useCatalogs(catalogs['hydra:member'])
  const contextValueMetadata = metadata['hydra:member']

  return (
    <OptionsProvider>
      <catalogContext.Provider value={contextValueCatalog}>
        <metadataContext.Provider value={contextValueMetadata}>
          {children}
        </metadataContext.Provider>
      </catalogContext.Provider>
    </OptionsProvider>
  )
}

export default TestProvider
