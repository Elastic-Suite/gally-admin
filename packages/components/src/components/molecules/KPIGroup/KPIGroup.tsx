import React from 'react'
import KPI, { IKPIPros } from '../../atoms/KPI/KPI'
import { Container } from './KPIGroup.styled'

interface IKPI extends IKPIPros {
  id: string | number
}

interface IKPIGroupProps {
  kpis: IKPI[]
}

function KPIGroup({ kpis }: IKPIGroupProps): JSX.Element {
  return (
    <Container>
      {kpis.map(({ id, ...value }) => (
        <KPI key={id} {...value} />
      ))}
    </Container>
  )
}

export default KPIGroup
