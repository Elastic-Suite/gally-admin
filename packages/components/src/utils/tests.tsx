import React, { ReactNode } from 'react'
import { RenderOptions, render, renderHook } from '@testing-library/react'
import { PreloadedState } from '@reduxjs/toolkit'
import { Bundle, api } from '@elastic-suite/gally-admin-shared'

import { AppStore, RootState, setupStore } from '../store'

import AppProvider from '../components/stateful-providers/AppProvider/AppProvider'

import TestProvider from './TestProvider'

export interface IExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: PreloadedState<RootState>
  store?: AppStore
}

export interface IRenderOutput {
  store: AppStore
}

const defaultState = {
  data: {
    api,
    bundles: [Bundle.VIRTUAL_CATEGORY],
    configurations: {
      'base_url/media': 'https://localhost/media/catalog/product/',
    },
    metadata: [
      {
        '@id': '/metadata/3',
        '@type': 'Metadata',
        id: 3,
        entity: 'product',
        sourceFields: ['/source_fields/217', '/source_fields/253'],
      },
      {
        '@id': '/metadata/4',
        '@type': 'Metadata',
        id: 4,
        entity: 'category',
        sourceFields: ['/source_fields/254', '/source_fields/286'],
      },
    ],
  },
  user: {
    requestedPath: '',
  },
}

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/explicit-function-return-type
export function renderWithProviders(
  ui: React.ReactElement,
  {
    preloadedState = defaultState,
    store,
    ...renderOptions
  }: IExtendedRenderOptions = {}
) {
  store = store ?? setupStore(preloadedState)
  function Wrapper({ children }: { children: ReactNode }): JSX.Element {
    return (
      <AppProvider store={store}>
        <TestProvider>{children}</TestProvider>
      </AppProvider>
    )
  }
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) }
}

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/explicit-function-return-type
export function renderHookWithProviders<R>(
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  hook: (props: any) => R,
  {
    preloadedState = defaultState,
    store,
    ...renderOptions
  }: IExtendedRenderOptions = {}
) {
  store = store ?? setupStore(preloadedState)
  function Wrapper({ children }: { children: ReactNode }): JSX.Element {
    return (
      <AppProvider store={store}>
        <TestProvider>{children}</TestProvider>
      </AppProvider>
    )
  }
  return { store, ...renderHook(hook, { wrapper: Wrapper, ...renderOptions }) }
}
