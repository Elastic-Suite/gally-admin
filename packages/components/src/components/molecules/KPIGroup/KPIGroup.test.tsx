import React from 'react'

import { renderWithProviders } from '../../../utils/tests'

import KPIGroup from './KPIGroup'

describe('KPIGroup match snapshot', () => {
  beforeAll(() => {
    jest
      .spyOn(Number.prototype, 'toLocaleString')
      .mockImplementation(function (this: number, _locale, options) {
        return new Intl.NumberFormat('fr-FR', options).format(this)
      })
  })

  afterAll(() => {
    jest.restoreAllMocks()
  })

  it('KPIGroup', () => {
    const { container } = renderWithProviders(
      <KPIGroup
        kpis={[
          {
            id: 1,
            label: 'Test 1',
            value: 456,
          },
          {
            id: 2,
            label: 'Test 2',
            value: 46.86,
            isPercentage: true,
          },
        ]}
      />
    )
    expect(container).toMatchSnapshot()
  })
})
