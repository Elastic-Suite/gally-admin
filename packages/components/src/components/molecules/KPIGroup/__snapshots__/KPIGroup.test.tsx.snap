// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KPIGroup match snapshot KPIGroup 1`] = `
<div>
  <div
    class="css-1l3liia"
  >
    <div
      class="css-4jtltt"
    >
      <h3
        class="css-1xva16q"
      >
        Test 1
      </h3>
      <p
        class="css-2384rf"
      >
        456
      </p>
    </div>
    <div
      class="css-4jtltt"
    >
      <h3
        class="css-1xva16q"
      >
        Test 2
      </h3>
      <p
        class="css-2384rf"
      >
        46,86%
      </p>
    </div>
  </div>
</div>
`;
